BD-3050-2 Update wording for user group sizes (#7696)
braze-inc/braze-docs: _docs/_user_guide/data_and_analytics/reporting/global_control_group_reporting.md

### Group sizes

> Before May 2024, the Global Control Group was excluded from user archival, but the treatment sample group was not. Starting May 2024, both groups were excluded from user archival. This could result in your treatment sample group and Global Control Group having significantly different sizes. The next time you reset your Global Control Group, this discrepancy will go away and you'll see similar group sizes.

Reviewer: ```suggestion
Before May 2024, the Global Control Group was excluded from user archival, but the treatment sample group was not. Starting May 2024, both groups are excluded from user archival. This could result in your treatment sample group and Global Control Group having significantly different sizes. The next time you reset your Global Control Group, this discrepancy will resolve and you'll see similar group sizes.
```
